priv/repo/migrations/20201110152910_create_notifications.exs

```diff
@@ -0,0 +1,34 @@
+defmodule Skate.Repo.Migrations.StoreNotifications do
+  use Ecto.Migration
+
+  def up do
+    create table(:notifications) do
+      add(:created_at, :bigint, null: false)
+      add(:reason, :string, null: false)
+      add(:route_ids, {:array, :string}, null: false)
+      add(:run_ids, {:array, :string}, null: false)
+      add(:trip_ids, {:array, :string}, null: false)
+      add(:operator_id, :string)
+      add(:operator_name, :string)
+      add(:route_id_at_creation, :string)
+      add(:block_id, :string, null: false)
+      add(:service_id, :string, null: false)
+      add(:start_time, :bigint, null: false)
+      add(:end_time, :bigint, null: false)
+      timestamps()
+    end
+
+    create(
+      index(
+        :notifications,
+        [:start_time, :end_time, :block_id, :service_id, :reason],
+        unique: true,
+        name: "notifications_unique_index"
+      )
+    )
+  end
+
+  def down do
```

Comment:
Was there a reason to use separate `up`/`down` functions instead of a single `change` function?

Reply:
There is--the index. If you have something like:

```
def change do
  create table :foo do
    # some columns defined
  end
  create index :foo, [:some_column, :other_column]
end
```

and you roll that back, it wants to drop the index before dropping the table. But dropping the index separately isn't necessary, since when you drop a table all the indices on it disappear too. 

It also doesn't really hurt to drop the index separately, except that during development, when I was tweaking the index, the separate-drop scheme led to some annoyances that were breaking my flow.